packages/radio-player/src/search-handler/search-handler.ts

```diff
@@ -149,6 +149,12 @@ export class SearchHandler implements SearchHandlerInterface {
     const searchRanges: Range[] = await this.searchBackend.getSearchRanges(query);
     const { mergedTranscript } = this.transcriptIndex;
 
+    // Sort the search range results from start to end
+    // The search engine may return results in any order but while processing the results here,
+    // we are going from start to finish. If the search results are out of order, it rebuilds
+    // incorrect blocks of search results.
+    searchRanges.sort((a: Range, b: Range) => (a.startIndex > b.startIndex ? 1 : -1));
```

Comment:
QUESTION: Doesn't `sort` also need an equality case, returning `0`? Won't cause any endless loops/probs if not?

Reply:
I updated the comparison to take into account the equality case, but since we have individual word matches in our search results, we shouldn't ever get two matching start indices in our results.

This did get me thinking of the case, though, if the search engine behavior changed and we did get overlapping search results. I think we'd need to figure out how to handle that in the UI. ie. Do we merge the search result ranges, break them up into individual results, etc. There's a few cases I could see us running into.